### ISSUE TITLE
attn_drop in GATConv

### ISSUE THREAD

**Issue by mwcvitkovic** [dmlc/dgl#1016]
## 🐛 Bug

I'm not 100% sure this is a bug, but in `nn.pytorch.conv.GATConv.forward`, `self.attn_drop` is applied after `edge_softmax`.  Shouldn't it be applied to the attention weights *before* the softmax?  

Applying the dropout after the softmax, as it is now, means the sum of the softmax probabilities of a node's neighbors will not sum to one.

**Comment by mwcvitkovic** [comment]
Oops, my apologies.  Now that I look at the [code that goes with the original paper,]( https://github.com/PetarV-/GAT/blob/ea3aeafb782e20738a090dbf7914aa6e5a1a137e/utils/layers.py) it looks like they do apply the dropout to the normalized weights. 

This still seems unusual to me though!
